kong/router.lua

```diff
@@ -41,6 +43,16 @@ local ERR           = ngx.ERR
 local WARN          = ngx.WARN
 
 
+local DEFAULT_HOSTNAME_TYPE = hostname_type("")
+
+
+local function append(destination, value)
+  local n = destination[0] + 1
```

Comment:
the common Lua paradigm for this would be to use a field called `n`. Any specific reason to not use `n` but using index `0`?